lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -527,10 +527,14 @@ sub schema_fixup
         if ($packaging) {
             if (ref($packaging) eq 'HASH') {
                 # MB Solr search server v3.1
-                $data->{packaging} = MusicBrainz::Server::Entity::ReleasePackaging->new(
+                $data->{packaging} = defined $packaging->{id}
+                ? MusicBrainz::Server::Entity::ReleasePackaging->new(
                     name => $packaging->{name},
                     gid => $packaging->{id}
```

Comment:
This works, but there's a simpler way to make a hash entry optional in Perl without duplicating the definition:
```suggestion
                     defined $packaging->{id} ? (gid => $packaging->{id}) : (),
```

Reply:
applied in https://github.com/metabrainz/musicbrainz-server/pull/1234/commits/c8577d080a9c4d27e19d41ab939017349ec11711